barretenberg/cpp/src/barretenberg/stdlib/primitives/bigfield/bigfield.test.cpp

```diff
@@ -443,10 +416,50 @@ template <typename Builder> class stdlib_bigfield : public testing::Test {
                 std::cout << "num gates per div = " << after - before << std::endl;
                 benchmark_info(Builder::NAME_STRING, "Bigfield", "DIV", "Gate Count", after - before);
             }
-            // uint256_t modulus{ Bn254FqParams::modulus_0,
-            //                    Bn254FqParams::modulus_1,
-            //                    Bn254FqParams::modulus_2,
-            //                    Bn254FqParams::modulus_3 };
+
+            fq expected = (inputs[0] / inputs[1]);
+            expected = expected.reduce_once().reduce_once();
+            expected = expected.from_montgomery_form();
+            uint512_t result = c.get_value();
+
+            EXPECT_EQ(result.lo.data[0], expected.data[0]);
+            EXPECT_EQ(result.lo.data[1], expected.data[1]);
+            EXPECT_EQ(result.lo.data[2], expected.data[2]);
+            EXPECT_EQ(result.lo.data[3], expected.data[3]);
+            EXPECT_EQ(result.hi.data[0], 0ULL);
+            EXPECT_EQ(result.hi.data[1], 0ULL);
+            EXPECT_EQ(result.hi.data[2], 0ULL);
+            EXPECT_EQ(result.hi.data[3], 0ULL);
+        }
+        bool result = CircuitChecker::check(builder);
+        EXPECT_EQ(result, true);
+    }
+
+    static void test_div_with_constant()
+    {
+        auto builder = Builder();
+        size_t num_repetitions = 10;
+        for (size_t i = 0; i < num_repetitions; ++i) {
+            fq inputs[2]{ fq::random_element(), fq::random_element() };
+            inputs[0] = inputs[0].reduce_once().reduce_once();
+            inputs[1] = inputs[1].reduce_once().reduce_once();
+
+            // a is witness, b is constant
+            fq_ct a(witness_ct(&builder, fr(uint256_t(inputs[0]).slice(0, fq_ct::NUM_LIMB_BITS * 2))),
+                    witness_ct(&builder,
+                               fr(uint256_t(inputs[0]).slice(fq_ct::NUM_LIMB_BITS * 2, fq_ct::NUM_LIMB_BITS * 4))));
+            fq_ct b(&builder, uint256_t(inputs[1]));
```

Comment:
Do we have a contant/witness case somewhere?

Reply:
Good, point. I added a test case of $\frac{\textsf{constant}}{\textsf{witness}}$.